core/domain/summary_services.py

```diff
@@ -146,6 +234,29 @@ def _get_displayable_exp_summary_dicts(exploration_summaries):
 
     return displayable_exp_summaries
 
+def _get_displayable_col_summary_dicts(col_summaries):
+    col_summaries_to_display = []
+    for col_summary in col_summaries:
+        if col_summary and col_summary.status != (
+                rights_manager.ACTIVITY_STATUS_PRIVATE):
+            col_summaries_to_display.append({
+                'id': col_summary.id,
+                'title': col_summary.title,
+                'category': col_summary.category,
+                'activity_type': rights_manager.ACTIVITY_TYPE_COLLECTION,
+                'objective': col_summary.objective,
+                'num_explorations': len(
```

Comment:
Please update ``CollectionSummary`` to include a ``node_count`` value and make sure it's updated in the correct places in ``collection_services`` rather than doing this. ``get_collection_by_id`` is a very expensive operation to do here (1 RPC to fetch an entire collection for each collection summary; ouch). Moreover, this function should be lightweight. It's basically converting a collection summary to a stripped down dict. Let's try and preserve that behavior for future developers who edit this code.